src/server/handshake.js

```diff
@@ -9,10 +9,14 @@ module.exports = function (client, server, { version }) {
     client.protocolVersion = packet.protocolVersion
 
     if (version === false || version === undefined) {
-      if (require('minecraft-data')(client.protocolVersion)) {
-        client.version = client.protocolVersion
+      if (client.protocolVersion === 5) { // The first snapshot versions of 1.8 uses the 1.7 protocol id, after 1.8 each snapshot has a different protocol id.
```

Comment:
what is the bug and why can't it be fixed in the data ?

Reply:
Incompatible protocol version that report the same protocolVersion. Can't be fixed in data because there's nothing to fix, both the snapshots and 1.7.10 version report proto version 5

Reply:
looking at https://github.com/PrismarineJS/minecraft-data/blob/master/data/pc/common/protocolVersions.json#L3008 ; what prevent us to take the last version for this protocol version and for this major version ?

Reply:
this should be fixed directly in node-minecraft-data version guessing from protocol version logic

Reply:
https://github.com/PrismarineJS/node-minecraft-data/blob/63e8769d8f3e71d7df0678773ac8e58f9b58c178/index.js#L62 there